src/artifacts.js

```diff
@@ -502,13 +502,27 @@ var replyWithArtifact = async function(taskId, runId, name, req, res) {
     return res.reportError('ResourceNotFound', 'Artifact not found', {});
   }
 
+  // Some downloading utilities need to know the artifact's storage type to be
+  // able to handle their downloads most correctly.  We're going to set this
+  // field on all artifact responses so that the downloading utilities can use
+  // slightly different logic for each artifact type
+  res.set('x-taskcluster-artifact-type', artifact.storageType);
+
   if (artifact.storageType === 'blob') {
     // Most of the time, the same base options are used.
     let getOpts = {
       bucket: artifact.details.bucket,
       key: artifact.details.key,
     };
 
+    res.set('x-taskcluster-content-sha256', artifact.details.contentSha256);
```

Comment:
These are headers for the thing we are redirecting to, hence, I suggest they are on the form:
`x-taskcluster-location-`

like `x-taskcluster-location-content-sha256` is the `content-sha256` of what the `location` header points to.